mathics/core/expression.py

```diff
@@ -1155,6 +1167,21 @@ def block(tex, only_subsup=False):
         else:
             raise BoxError(self, 'tex')
 
+    def output_cost(self):
+        name = self.get_head_name()
+
+        if name in ('System`ImageBox', 'System`GraphicsBox', 'System`Graphics3DBox'):
+            return 0  # always display
+
+        cost = sum(leaf.output_cost() for leaf in self.leaves)
+
+        if name == 'System`List':
+            return 2 + cost + len(self.leaves)  # {a, b, c}
+        elif name in _layout_boxes:
+            return cost
+        else:
+            return cost + 2 + self.head.output_cost() + len(self.leaves)  # XYZ[a, b, c]
```

Comment:
`cost` is the sum cost of leaves (I renamed it now). `len(self.leaves)` measures the size of ",", which could be one or two (it assumes for now). we could pass the separator size into `make_boxes` and then pass it on to `output_cost`, but it's bulky and then should we really measure it that way? If `$OutputSizeLimit` is understood as number of printable characters (without spaces and layout elements like fractions, parentheses, ...), then it makes sense to always measure 1 for ",".